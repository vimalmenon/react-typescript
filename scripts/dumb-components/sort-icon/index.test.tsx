import "jsdom-global/register";
import * as React from "react";
import * as renderer from "react-test-renderer";
import SortIcon from "./index";
import {shallow} from "enzyme";


describe("SortIcon component :", () => {

    test("Snapshot renders :", () => {
        const component = renderer.create(<SortIcon sortIndex={1} currentIndex={1}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("SortIcon nothing to render :", () => {
        const wrapper = shallow(<SortIcon sortIndex={2} currentIndex={1}/>);
        expect(wrapper.isEmptyRender()).toBe(true);
    });

    test("SortIcon to render on ascending icon :", () => {
        const wrapper = shallow(<SortIcon sortIndex={2} currentIndex={2}/>);
        expect(wrapper.contains(<span>&darr;</span>)).toBe(true);
    });

    test("SortIcon to render on descending icon :", () => {
        const wrapper = shallow(<SortIcon sortIndex={-2} currentIndex={2}/>);
        expect(wrapper.contains(<span>&uarr;</span>)).toBe(true);
    });
    
});