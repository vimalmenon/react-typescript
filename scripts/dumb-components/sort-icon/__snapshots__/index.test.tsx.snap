// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortIcon component : Snapshot renders : 1`] = `
<span>
  â†“
</span>
`;
